shards/shards.go

```diff
@@ -238,33 +235,55 @@ func selectRepoSet(shards []rankedShard, q query.Q) ([]rankedShard, query.Q) {
 			continue
 		}
 
+		// setSize may be larger than the number of shards we have. The size of
+		// filtered is bounded by min(len(set), len(shards))
+		if setSize > len(shards) {
+			setSize = len(shards)
+		}
+
 		filtered := make([]rankedShard, 0, setSize)
 
 		for _, s := range shards {
-			if repositorer, ok := s.Searcher.(repositorer); ok {
-				repo := repositorer.Repository()
-				if hasRepo(repo.Name) {
-					filtered = append(filtered, s)
-				}
+			if hasRepo(s.name) {
+				filtered = append(filtered, s)
```

Comment:
I don't know the context of the code well enough to know this is equivalent, but I'm sure you do :)

Reply:
Yeah it is the same. This field didn't exist when this code was originally written hence the more roundabout route it took before.